### ISSUE TITLE
why the loss of generator only contains feature loss and style loss?

### ISSUE THREAD

**Issue by crissallan** [MengZephyr/HairGANs#5]
Hi, I was kind of confused why the self.G_loss only contains the feature loss and the style loss? 
If so, i think it might not be a GAN's advresarial training process, because you only update the parameters of the Generator based on the feature loss and style loss but not the adversarial loss.

**Comment by MengZephyr** [comment]
The GAN model used here is based on WGAN-GP. 
The adversarial loss please find the line:

 self.D_loss = tf.reduce_mean(self.D_fake)-tf.reduce_mean(self.D_real) + grad_pen

**Comment by MengZephyr** [comment]
> Hi, I was kind of confused why the self.G_loss only contains the feature loss and the style loss?
> If so, i think it might not be a GAN's advresarial training process, because you only update the parameters of the Generator based on the feature loss and style loss but not the adversarial loss.

The GAN model used here is based on WGAN-GP.
The adversarial loss please find the line:

self.D_loss = tf.reduce_mean(self.D_fake)-tf.reduce_mean(self.D_real) + grad_pen
